spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -164,15 +178,81 @@ private String createMessagingErrorMessage(String description, Object payload) {
 	/**
 	 * Delegates payload extraction to
 	 * {@link #extractMessage(org.springframework.amqp.core.Message message)}
-	 * to enforce backward compatibility.
+	 * to enforce backward compatibility. Uses this listener adapter's converter instead of
+	 * the one configured in the converter adapter.
+	 * If the inbound message has no type information and the configured message converter
+	 * supports it, we attempt to infer the conversion type from the method signature.
 	 */
 	private class MessagingMessageConverterAdapter extends MessagingMessageConverter {
 
+		private final Object bean;
+
+		private final Method method;
+
+		private final Type inferredArgumentType;
+
+		public MessagingMessageConverterAdapter(Object bean, Method method) {
```

Comment:
The `RedundantModifierChecker` will say us that we are bad here: the class is `private`.